pymatgen/io/vasp/sets.py

```diff
@@ -498,7 +511,7 @@ def nelect(self) -> float:
             return nelect
 
     @property
-    def kpoints(self) -> Kpoints:
+    def kpoints(self) -> Kpoints or None:
```

Comment:
This is not a valid type annotation, it should be `Optional[Kpoints]`

Reply:
changing to `Optional[Kpoints] is triggering a `NameError`

Reply:
You may need to import `Optional` if it's not already imported, the `or` is definitely incorrect though (to my knowledge). The `Optional` syntax is equivalent to `Union[None, Kpoints]`.

Reply:
Any objection to using `Union[Kpoints,None]` instead? That's more explicit and `Union` is already imported

Reply:
`Optional[Kpoints]` is definitely the more pythonic.

Reply:
@utf my understanding is that `Optional` is really intended for the case of optional arguments that default to None. We don't have an optional argument here - None has the specific meaning that kpoints are being determine by VASP.

Reply:
Ah, actually that's not the case. "Optional" has a specific meaning in type theory, https://en.wikipedia.org/wiki/Option_type It's not intended for "optional arguments" in the colloquial sense of the word

Reply:
Guess I shouldn't believe everything [I read on the internet](https://realpython.com/python-type-checking/#the-optional-type).

Sounds like Optional is preferred then?

Reply:
Yeah, let's go with Optional. If it's helpful, consider that a function might *return* an Optional type. In this case there was no "optional argument," nothing was chosen by the user, it's the type itself that is optional. In terms of resources for this, PEP 484 is the canonical source of information, and that doesn't take an explicit stance.